tests/version/test_markers.py

```diff
@@ -2154,6 +2154,35 @@ def test_complex_intersection() -> None:
     )
 
 
+def test_complex_union_is_deterministic() -> None:
+    """
+    This test might fail sporadically if marker operations are not deterministic!
```

Comment:
Does that mean it can fail because of anything, or will it detect changes to the marker operations code that would make them undeterministic?

Reply:
If the marker operations code becomes undeterministic there will be a good chance that the test fails sporadically. As long as the marker operations code is determistic the test will not fail.